unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/VideoPlayer/VideoPlayerRegister.cs

```diff
@@ -14,7 +14,9 @@ public VideoPlayerRegister(int componentId, ECSComponentsFactory factory, IECSCo
         {
             factory.AddOrReplaceComponent(componentId,
                 ProtoSerialization.Deserialize<PBVideoPlayer>,
-                () => new VideoPlayerHandler(internalComponents.videoPlayerComponent, DataStore.i.Get<DataStore_LoadingScreen>().decoupledLoadingHUD));
+                () => new VideoPlayerHandler(
+                    internalComponents.videoPlayerComponent,
+                    DataStore.i.Get<DataStore_LoadingScreen>().decoupledLoadingHUD));
```

Comment:
can you inject the DataStore instance instead of coupling the singleton?

Reply:
we don't want to add dependency injection for those register classes.
long story short, they are the classes that setup the component, they are used only in prod code or for integration tests where we need to use the actual reference.
unit-test are created only for the handler classes, that's why it is injected for `VideoPlayerHandler` constructor, because in those cases we do want to have more control over that reference